graylog2-server/src/main/java/org/graylog2/inputs/InputServiceImpl.java

```diff
@@ -282,7 +282,8 @@ public void addStaticField(Input input, final String key, final String value) th
             // SOFT MIGRATION: does this extractor have an order set? Implemented for issue: #726
             Long order = 0L;
             if (ex.containsField(Extractor.FIELD_ORDER)) {
-                order = (Long) ex.get(Extractor.FIELD_ORDER); // mongodb driver gives us a java.lang.Long
+                Number num = (Number) ex.get(Extractor.FIELD_ORDER);
+                order = num.longValue(); // mongodb driver gives us a java.lang.Long
```

Comment:
As mentioned in the commit message:
> To use the grok pattern extractor in the inputs.json
> this change changes the conversion from number to long.
> Since the mongodb driver will return the order as long
> only a cast was necessary. But now I use a sample
> db in json. And json does not support long. So I cast
> the order value (which is only -1, 0, 1) to Number and
> than convert it to long.

Reply:
Sorry, I missed that. Can you add a comment to the code so we remember that in the future? :smiley: Thank you!

Reply:
With pleasure!